klippy/extras/dockable_probe.py

```diff
@@ -0,0 +1,596 @@
+            self._align_to_vector(self.dock_angle)
+        retry = 0
+        while (self.get_probe_state() != PROBE_ATTACHED
+               and retry < self.dock_retries):
+            self._do_attach()
+            retry += 1
+        if self.get_probe_state() != PROBE_ATTACHED:
+            raise self.printer.command_error('Probe attach failed!')
+        self.post_attach_gcode.run_gcode_from_command()
+        if return_pos:
+            if not self._check_distance(return_pos, self.approach_distance):
+                self.toolhead.manual_move(
+                    [return_pos[0], return_pos[1], None],
+                    self.travel_speed)
+    def _do_attach(self):
+        if self.get_probe_state() != PROBE_DOCKED:
+            raise self.printer.command_error(
+                'Attach Probe: Probe not detected in dock, aborting')
+        if self._check_distance(dist=self.approach_distance):
+            self._align_to_vector(self.dock_angle)
+        else:
+            self._move_to_vector(self.dock_angle)
+        self.toolhead.manual_move(
+            [self.dock_position[0],self.dock_position[1], None],
+             self.attach_speed)
+        self.attach_gcode.run_gcode_from_command()
+        if self._check_distance(dist=self.approach_distance):
+            self._align_to_vector(self.dock_angle)
+        self.toolhead.manual_move(
+            [self.approach_position[0], self.approach_position[1], None],
+             self.travel_speed)
+
+    # Detaching actions
+    def detach_probe(self, return_pos=None):
+        self.pre_detach_gcode.run_gcode_from_command()
+        if self.get_probe_state() == PROBE_ATTACHED:
+            self._align_z()
+        self.toolhead.manual_move(
+            [self.approach_position[0], self.approach_position[1], None],
+             self.travel_speed)
+        retry = 0
+        while (self.get_probe_state() != PROBE_DOCKED
+               and retry < self.dock_retries):
+            self._do_detach()
+            retry += 1
+        if self._check_distance(dist=self.detach_distance):
+            self._align_to_vector(self.detach_angle)
+            self.toolhead.manual_move(
+                [self.detach_position[0], self.detach_position[1], None],
+                 self.travel_speed)
+        if self.get_probe_state() != PROBE_DOCKED:
+            raise self.printer.command_error('Probe detach failed!')
+        self.post_detach_gcode.run_gcode_from_command()
+        if return_pos:
+            if not self._check_distance(return_pos, self.detach_distance):
+                self.toolhead.manual_move(
+                    [return_pos[0], return_pos[1], None],
+                    self.travel_speed)
+    def _do_detach(self):
+        if self._check_distance(dist=self.detach_distance):
+            self._align_to_vector(self.dock_angle)
+        else:
+            self.toolhead.manual_move(
+                [self.detach_position[0], self.detach_position[1], None],
```

Comment:
I believe this should be approach_position (otherwise you're going detach -> dock -> detach, not approach -> dock -> detach). 
```suggestion
                [self.approach_position[0], self.approach_position[1], None],
```

Reply:
👍 I was also having problems with detach, though strangely only when the `x` detach position was the same as the `x` attach position.

Either way, the behavior was `detach_position` -> `dock_position` -> `detach_position`, which is not correct.

Making this change solved my problems.

Reply:
I was helping another user on discord the other day when we discovered this typo. I should have to patched out in the next day or so along with some other corrections.